halibot/halobject.py

```diff
@@ -73,13 +73,30 @@ def sync_send_to(self, msg, dests):
 
 	async def _receive(self, msg):
 		try:
-			self.receive(msg)
+			fname = 'receive_' + msg.type
+			if hasattr(self, fname) and callable(getattr(self, fname)):
```

Comment:
I really wish Python let you use assignment as a boolean expression here. I'm torn between caching the `getattr` call so we don't have to do a second one, and keeping the simplicity and cleaniness of this.